### PR TITLE
3.get/api/categories

### PR DESCRIPTION
**Pull request by mini-wizbit**
Starting on 3.GET/api/categories
check the commit as time goes on

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const db = require("../db/connection.js");
+const { app } = require("../app.js");
+const request = require("supertest");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data/");
+const { response } = require("express");
```
Comment (pkaodev) [inline __tests__/app.test.js]: don't need this

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+afterAll(() => {
+  db.end();
```
Comment (pkaodev) [inline __tests__/app.test.js]: consider the asynchronous nature of this method.  it returns a promise, what do we have to do with promises?

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+beforeEach(() => {
+  return seed(testData); // <- seeding each time DELETE COMMENT later!
```
Comment (pkaodev) [inline __tests__/app.test.js]: let's delete this comment

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+describe("Using app.js to run the database of NC-games", () => {
```
Comment (pkaodev) [inline __tests__/app.test.js:15]: -dersThink about the structure of describe/test blocks.  Is this first describe helping us understand what the tests are doing?

There is no prescribed way of doing it, but an example could be:
describe("GET api/categories")
test("200: returns array of category objects")

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  describe("GET api/categories will return an array of 'category' objects with the following properties: slug and description", () => {
+    test("when GET api/categories is called returns a array of objects", () => {
+      return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then((response) => {
+          expect(Array.isArray(response.body.categories)).toBe(true);
+          expect(response.body.categories.length > 0).toBe(true);
```
Comment (pkaodev) [inline __tests__/app.test.js:23]: What kind of objects are we expecting to be returned?

File: app.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+const app = express();
+
+// custom error <- DELETE this comment later on
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send(err);
+  } else {
+    next();
+  }
+});
```
Comment (pkaodev) [inline app.js]: We aren't working on error-handling yet - please remove for now

**Comment by pkaodev** [comment]
Great work so far.

You are missing the actual endpoint, please only submit a PR when you have completed a ticket.

Remove rogue console.logs

Testing for this endpoint needs to be more complete.

Please submit another PR with changes.

File: __tests__/app.test.js (hunk 1 above)
Comment (mini-wizbit) [reply __tests__/app.test.js]: Removed

File: __tests__/app.test.js (hunk 2 above)
Comment (mini-wizbit) [reply __tests__/app.test.js]: we need to return

File: __tests__/app.test.js (hunk 3 above)
Comment (mini-wizbit) [reply __tests__/app.test.js]: done

File: __tests__/app.test.js (hunk 4 above)
Comment (mini-wizbit) [reply __tests__/app.test.js:15]: Okay that's understandable

File: __tests__/app.test.js (hunk 1 above)
Comment (mini-wizbit) [reply __tests__/app.test.js]: Found out it was only the first one

File: __tests__/app.test.js (hunk 5 above)
Comment (mini-wizbit) [reply __tests__/app.test.js:23]: Was originally just gonna test for the input/output
then build up with the simple behaviours after but if i can do it all in one then I;m doing it

File: app.js (hunk 6 above)
Comment (mini-wizbit) [reply app.js]: Sorry thats me just defaulting with a set up and a custom error

File: Controllers/categories.controller.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const { selectCategories } = require("../Model/categories.model");
+
+exports.getCategories = (req, res, next) => {
+  selectCategories("this")
```
Comment (pkaodev) [inline Controllers/categories.controller.js:4]: What is "this"?  selectCategories doesn't take an argument according to how you've defined it.

**Comment by pkaodev** [comment]
Have a look at the comment on how you're using selectCategories in your controller.   Have a look at that, and then you are free to merge (no additional PR needed)

File: Controllers/categories.controller.js (hunk 7 above)
Comment (mini-wizbit) [reply Controllers/categories.controller.js:4]: Was a placholder for temp purposes for when I came back out of the model
has been Adjusted since

**Pull request merged**